Classes/AbstractClient.php

```diff
@@ -44,36 +45,9 @@ protected function getTranslator(): Translator
         if ($this->configuration->getApiKey() === '') {
             throw new ApiKeyNotSetException('The api key ist not set', 1708081233823);
         }
-        $proxyUrl = $this->getConfiguredSystemProxy();
-        $options = [];
-        if ($proxyUrl !== null) {
-            $options[TranslatorOptions::PROXY] = $proxyUrl;
-        }
+        $options[TranslatorOptions::HTTP_CLIENT] = GeneralUtility::makeInstance(GuzzleClientFactory::class)->getClient();
```

Comment:
In TYPO3 v11 `GuzzleClientFactory::getClient()` is static, that changed in v12. However, it is safe to call it the non-static way in both cases on a class instance [1].

It would be properly better to get the factory or injected, but I would take that for now this way and streamline that with the next version (dropping v11 support).
And we can think with time if we want directly a client instance injected instead of the factory along with some streamlining. Thus, postponing making this DI aware right now.

Setting a full client is supported by the DeepL PHP API [2], which make this a doable replacement.

Would make sense to add additional tests to the line with the DI change when mockedfactory can be used.


[1] See https://3v4l.org/a8RI2
[2] See https://github.com/DeepLcom/deepl-php?tab=readme-ov-file#custom-http-client